repository_service_tuf/cli/artifact/repository.py

```diff
@@ -0,0 +1,261 @@
+# SPDX-FileCopyrightText: 2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+
+import base64
+
+from click import Context
+from dynaconf.loaders.yaml_loader import write  # type: ignore
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+
+
+def write_config(settings_path, settings_data, merge=True):
+    write(settings_path, settings_data, merge)  # pragma: no cover
+
+
+@artifact.group()
+@click.pass_context
+def repository(context) -> None:
+    """
+    Repository management.
+    """
+
+
+@repository.command()
+@click.option(
+    "-a",
+    "--all",
+    help="Show all the configured repositories.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+@click.argument("repository", required=False)
+def show(
+    context: Context,
+    all: bool,
+    repository,
+) -> None:
+    """
+    List configured repositories.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if all:
+        if rstuf_config.get("REPOSITORIES"):
+            for repo in rstuf_config.get("REPOSITORIES"):
+                console.print(repo)
+        else:
+            console.print("There are no configured repositories")
+            return
+
+    if repository:
+        if rstuf_config.get("REPOSITORIES") and rstuf_config[
+            "REPOSITORIES"
+        ].get(repository):
+            console.print(rstuf_config["REPOSITORIES"].get(repository))
+        else:
+            console.print(
+                f"Repository {repository} is missing in your configuration"
+            )
+
+
+@repository.command()
+@click.pass_context
+@click.argument("repository")
+def use(context: Context, repository) -> None:
+    """
+    Switch current repository.
+    """
+
+    rstuf_config = context.obj.get("settings").as_dict()
+
+    rstuf_config["CURRENT_REPOSITORY"] = repository
+    if context.obj.get("config"):
+        write_config(context.obj.get("config"), rstuf_config, False)
+        console.print(f"Current repository changed to {repository}")
+
+
+@repository.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+@click.argument("repository")
+def set(
+    context: Context,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+    repository,
```

Comment:
Nit: Isn't it better for the `repository` to be an explicit parameter `--name` in the `set` (I would prefer `add` as I mentioned above)

```suggestion
    repository: str,
```

Reply:
We previously agreed on `rstuf artifact repository add <repo_name>`, but if you prefer a flag, I don't mind changing the layout.

Reply:
I still think that having `add [REPOSITORY]` is confusing for the user
![screenshot_2024-01-25_at_18 06 54_720](https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/546882/8e4cc21e-010d-4874-8e2d-76a247e063c9)
t leads the user to add an empty repository of information :confused:
Another thing I noticed is that I don't have a way to check my settings without opening the file
![screenshot_2024-01-25_at_18 07 37_480](https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/546882/0655f5c0-61af-48d4-8b5c-43eb54568b24)


Another thing is that the `rstuf admin artifacts repository` should list all existent repository names and the default one should be highlighted.

 Otherwise, I need to remember my repositories' names.